### PR TITLE
Passing HostConfig for /start endpoint

### LINKED ISSUE #1781: Start container command not forwarding the HostConfig to the container
Client tools (Docker plugin for Jenkins in our case) that rely on passing (additional) HostConfig in the start container command do not work since 1.1.0. For example the port bindings are not passed and the container is inaccessible.

I believe the commit for 'Add support for container rescheduling on node failure.' (https://github.com/docker/swarm/commit/13f60212f57dae8ba480c28bf26261bd7331ea4) broke this flow.

From the Docker Remote API docs:
> 
Start a container:
For backwards compatibility, this endpoint accepts a HostConfig as JSON-encoded request body.
Reference: https://docs.docker.com/engine/reference/api/docker_remote_api_v1.22/#start-a-container

### PR DESCRIPTION
**Pull request by nishanttotla**
Fixes #1781. `HostConfig` is being deprecated for the `/start` endpoint, but we must support it until Engine does.

Signed-off-by: Nishant Totla <nishanttotla@gmail.com>

### CODE REVIEW

**Comment by nishanttotla** [comment]
Ping @vieux @abronan @dongluochen some tests are failing and they're related, I'm not sure why. It would be really helpful if you could help review the code changes, since this issue is on the 1.1.1 Milestone.

File: api/handlers.go [hunk 1]
```diff
@@ -747,6 +754,8 @@ func postContainersStart(c *context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	container.Config.HostConfig = hostConfig
```
Comment (vieux) [inline api/handlers.go]: you are erasing the hostconfig given at create, i thik we need to merge them instead.

Comment (vieux) [inline api/handlers.go]: Or, as we did before, don't even save the config on start, just pass it via are to dockerclient, I think it makes sense

**Comment by nishanttotla** [comment]
Closing PR since I made a git mess. Reopening a new one instead, sorry about that!

File: api/handlers.go (hunk 1 above)
Comment (nishanttotla) [inline api/handlers.go]: Thanks @vieux, this makes sense. Sorry I have to close this PR due to some git issues, let's move the conversation over to #1829.

**Pull request closed without merging**